packages/react-core/src/Authenticator/hooks/useAuthenticatorInitMachine/useAuthenticatorInitMachine.tsx

```diff
@@ -0,0 +1,20 @@
+import React from 'react';
+import { AuthenticatorMachineOptions } from '@aws-amplify/ui';
+
+import { useAuthenticator } from '../useAuthenticator';
+
+// Utility hook that sends init event to the parent provider
+export default function useAuthenticatorInitMachine(
+  data: AuthenticatorMachineOptions
+): void {
+  const { route, initializeMachine } = useAuthenticator(({ route }) => [route]);
```

Comment:
Not like a huge deal or anything but we could declare the selector param outside the scope of `useAuthenticatorInitMachine` to give it a static reference between renders

Reply:
"React Hook `useAuthenticator` cannot be called at the top level. React Hooks must be called in a React function component or a custom React Hook function."